### DISCUSSION TITLE
Tailwind CSS support Deno

### DISCUSSION THREAD

**Discussion by mojtabakaviani** [tailwindlabs/tailwindcss#2971] [Ideas]
Please add Tailwind CSS to [Deno](https://deno.land/x) Modules and [Postcss](https://github.com/postcss/postcss-deno-import) transform.

**Comment by K-REBO** [comment]
How about [Deno PostCss plugin](https://deno.land/x/postcss@8.1.6)

**Comment by aadamsx** [comment]
Is Tailwind compatible with Deno + React?

**Reply by eliassjogreen** [reply]
Doesn't work for me as I am getting the following error when using the tailwindcss plugin for postcss:
```ts
import postcss from "https://esm.sh/postcss@8.2.8";
import tailwindcss from "https://esm.sh/tailwindcss@2.0.4?external=postcss@8.2.8";

const result = await postcss([tailwindcss]).process(css);
```
throws
```
error: Uncaught TypeError: g_.Parser.extend(...).extend is not a function
    at https://cdn.esm.sh/v29/tailwindcss@2.0.4/external=postcss@8.2.8/deno/tailwindcss.js:25:17336
    at https://cdn.esm.sh/v29/tailwindcss@2.0.4/external=postcss@8.2.8/deno/tailwindcss.js:2:759
    at https://cdn.esm.sh/v29/tailwindcss@2.0.4/external=postcss@8.2.8/deno/tailwindcss.js:25:26231
    at https://cdn.esm.sh/v29/tailwindcss@2.0.4/external=postcss@8.2.8/deno/tailwindcss.js:2:759
    at https://cdn.esm.sh/v29/tailwindcss@2.0.4/external=postcss@8.2.8/deno/tailwindcss.js:25:27733
    at https://cdn.esm.sh/v29/tailwindcss@2.0.4/external=postcss@8.2.8/deno/tailwindcss.js:2:759
    at https://cdn.esm.sh/v29/tailwindcss@2.0.4/external=postcss@8.2.8/deno/tailwindcss.js:87:36739
    at https://cdn.esm.sh/v29/tailwindcss@2.0.4/external=postcss@8.2.8/deno/tailwindcss.js:2:759
    at https://cdn.esm.sh/v29/tailwindcss@2.0.4/external=postcss@8.2.8/deno/tailwindcss.js:87:37941
```

**Reply by ghost** [reply]
Is that a [acorn](https://github.com/acornjs/acorn) problem then?
![image](https://user-images.githubusercontent.com/10061147/114269255-3be0c680-9a06-11eb-8bd9-aeaf13417d8f.png)

I used these imports:
```
import postcss from 'https://deno.land/x/postcss/mod.js'
import tailwindcss from 'https://jspm.dev/tailwindcss'
```

**Reply by lucidNTR** [reply]
its more than acorn. my list is currently (im sure there is more):  acorn-node fsevents chokidar. we need a tailwind module export that does not make use of node modules or direct filesystem apis...   im currently working on bundling a reexport of tailwind with rollup and try to inject stubs or node-compat libraries for those

**Reply by lucidNTR** [reply]
update: it seems to really boil down to acorn compatibility. the other modules could be polyfilled so far, but also with the deno node-polifills and the nodePolifill plugin of rollup i do not get past this error:

error: Uncaught TypeError: acorn.Parser.extend(...).extend is not a function

**Reply by tzvc** [reply]
@lucidNTR Were you able to polyfill acorn?

**Comment by lucidNTR** [comment]
nope, I had to migrate to the tailwind compatible windicss in the end which
had no such issues and served me well since the

**Reply by kyeotic** [reply]
How did you get this setup?

**Reply by lucidNTR** [reply]
i wrote a custom build plugin inspired loosely based on https://github.com/windicss/svelte-windicss-preprocess

what you need to do is hugely dependent on where it should be integrated and the rest of your pipeline
